python-checks/src/main/java/org/sonar/python/checks/hotspots/DebugModeCheck.java

```diff
@@ -21,52 +21,111 @@
+    if (!(callExpression.callee() instanceof QualifiedExpression) || arguments.isEmpty()) {
+      return;
+    }
+
+    if (DJANGO_CONFIGURE_FQN.equals(getQualifiedName(callExpression))) {
+      arguments.stream().filter(DebugModeCheck::isDebugArgument).forEach(arg -> ctx.addIssue(arg, MESSAGE));
+    }
+
+    if (FLASK_RUN_FQN.equals(getQualifiedName(callExpression))) {
+      RegularArgument debugArgument = TreeUtils.nthArgumentOrKeyword(2, "debug", arguments);
+      Optional.ofNullable(debugArgument)
+        .map(RegularArgument::expression)
+        .flatMap(TreeUtils.toOptionalInstanceOfMapper(Name.class))
+        .map(Name::name)
+        .filter(TRUE_KEYWORD::equals)
+        .ifPresent(str -> ctx.addIssue(debugArgument, MESSAGE));
+    }
+  }
+
+  private static void assignmentStatementConsumer(SubscriptionContext ctx) {
+    AssignmentStatement assignmentStatementTree = (AssignmentStatement) ctx.syntaxNode();
+    for (ExpressionList lhsExpression : assignmentStatementTree.lhsExpressions()) {
+      boolean isModifyingDebugProperty = isFlaskDebugProperties(lhsExpression)
+        || isGlobalDebugProperties(lhsExpression, ctx);
+      if (isModifyingDebugProperty && isTrueLiteral(assignmentStatementTree.assignedValue())) {
+        ctx.addIssue(assignmentStatementTree, MESSAGE);
       }
-    });
+    }
+  }
+
+  private static boolean isGlobalDebugProperties(ExpressionList expressionList, SubscriptionContext ctx) {
+    return expressionList.expressions().stream().anyMatch(DebugModeCheck::isDebugIdentifier)
+      && settingFiles.contains(ctx.pythonFile().fileName());
+  }
+
+  private static boolean isFlaskDebugProperties(ExpressionList expressionList) {
+    return expressionList.expressions().stream().anyMatch(DebugModeCheck::isModifyingFlaskDebugProperty);
+
+  }
+
+  private static boolean isModifyingFlaskDebugProperty(Expression expression) {
+    if (expression.is(Kind.QUALIFIED_EXPR)) {
+      return Optional.of((QualifiedExpression) expression)
+        .map(QualifiedExpression::symbol)
+        .map(Symbol::fullyQualifiedName)
+        .filter(FLASK_APP_DEBUG_FQN::equals)
+        .isPresent();
+    } else if (expression.is(Kind.SUBSCRIPTION)) {
+      SubscriptionExpression subscriptionExpression = (SubscriptionExpression) expression;
+      return Optional.of(subscriptionExpression)
+        .map(SubscriptionExpression::object)
+        .flatMap(TreeUtils.toOptionalInstanceOfMapper(QualifiedExpression.class))
+        .map(QualifiedExpression::symbol)
+        .map(Symbol::fullyQualifiedName)
+        .filter(FLASK_APP_CONFIG_FQN::equals)
+        .isPresent() && Optional.of(subscriptionExpression.subscripts())
+        .map(ExpressionList::expressions)
+        .filter(list -> list.size() == 1)
+        .map(list -> list.get(0))
+        .flatMap(TreeUtils.toOptionalInstanceOfMapper(StringLiteral.class))
+        .map(StringLiteral::trimmedQuotesValue)
+        .filter("DEBUG"::equals)
+        .isPresent();
```

Comment:
Maybe here we could refactor the two parts of the boolean into functions just to make it clearer about the intent. i.e. `isFlaskAppConfig` && `hasDebugModeEnabled` or something like that.

Reply:
Made the change in b19304898ea7456a2094bfd4607017b0420f105d and further in a34fd1b446737eb707412e6fc2427cc559630607.